### PR TITLE
Don't create a guest login if user went to /login

### PR DESCRIPTION
**Pull request by richvdh**
This fixes an unintuitive behaviour where, if you follow a link to
riot.im/app/#/login, we take you to the login page, but not before we've
registered a guest account (or restarted the MatrixClient with the stored
creds).

### CODE REVIEW

**Comment by richvdh** [comment]
Tests are currently failing because they need updating (ironically, they are waiting for the /sync to happen during startup)

File: src/components/structures/MatrixChat.js [hunk 1]
```diff
@@ -946,6 +953,7 @@ module.exports = React.createClass({
                 this.state.screenAfterLogin.screen,
                 this.state.screenAfterLogin.params,
             );
+            // XXX: is this necessary? `showScreen` should do it for us.
```
Comment (lukebarnard1) [inline src/components/structures/MatrixChat.js]: Actually, it's not necessary

Comment (richvdh) [inline src/components/structures/MatrixChat.js]: That's what I thought, but I didn't really want to change it at the same time :)

Comment (lukebarnard1) [inline src/components/structures/MatrixChat.js]: Fair enough!

**Pull request merged**